app/src/main/java/com/duckduckgo/mobile/android/duckduckgo/data/base/JsonEntity.java

```diff
@@ -0,0 +1,17 @@
+package com.duckduckgo.mobile.android.duckduckgo.data.base;
+
+/**
+ * Created by fgei on 6/12/17.
+ */
+
+public abstract class JsonEntity {
```

Comment:
No an interface won't force an empty constructor. If you get an early failure (e.g an exception is thrown) and you write a test to demand an empty constructor for each entity, then I'm ok with that. If you really want to keep it abstract go for it, it's a compromise either way.

Another option you have is rather than passing the class type into shared preferences, you could pass a lightweight object responsible for instantiating the entity. This would internally knows the correct type so you would need to build one for each type. That  keeps the code flexible and removes your dependence on a empty constructor.